x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/conditional/Greatest.java

```diff
@@ -152,7 +152,7 @@ public ExpressionEvaluator.Factory toEvaluator(ToEvaluator toEvaluator) {
         if (dataType == DataType.INTEGER) {
             return new GreatestIntEvaluator.Factory(source(), factories);
         }
-        if (dataType == DataType.LONG || dataType == DataType.DATETIME) {
+        if (dataType == DataType.LONG || dataType == DataType.DATETIME || dataType == DataType.DATE_NANOS) {
```

Comment:
I guess that having a DataType method to check on the two date types would be useful (a la `isDateTimeOrNanos()`).

Reply:
I really don't like adding more of those "helper" functions.  In my opinion, it's not any more readable than just having another case in the switching logic, and in fact may be less readable.  And I'd like to refactor this to use a map anyway, which we could reuse in the type checker.